File: java/src/automation-root/automation-core/src/main/java/com/automation/core/logging/TraceLogger.java
package com.automation.core.logging;

public class TraceLogger implements Logger {
    public void debug(String message) {
        System.out.println(message);
    }

    public void debug(String format, Object... args) {
        System.out.println(String.format(format, args));
    }

    public void debug(Exception exception, String message) {
        System.out.println(message + " " + exception.toString());
    }
}

File: csharp/src/AutomationRoot/Automation.Core/Logging/TraceLogger.cs
using System;
using System.Diagnostics;

namespace Automation.Core.Logging
{
    public class TraceLogger : ILogger
    {
        public void Debug(string message)
        {
            Trace.TraceInformation(message);
        }

        public void Debug(string format, params object[] args)
        {
            Trace.TraceInformation(string.Format(format, args));
        }

        public void Debug(Exception exception, string message)
        {
            Trace.TraceError(message + " " + $"{exception}");
        }
    }
}
